Implement CSV extraction from binary stream and format for embeddings

xichen-ansearch opened the pull request:
Hi,

I've implemented the following features and improvements:

  - Extracted CSV data from binary streams
  - Fetched and extracted CSV data directly from URLs
  - Integrated data cleaning within the extraction functions
  - Developed embedding functionalities for processed CSV data
  - Added unit and integration test

This PR is to close #17

Mael-Abgrall reviewed (changes requested):
Hi,

sorry, it seems we did not communicate well enough.

You do not need to do embedding, this is handled at a later stage

[look at the diagram of the application](https://github.com/A-star-logic/memoire/blob/main/docs/development/memoire%20architecture.png)

What I would like you to do is very very simple (much more than what you did).
You should only take a stream as an input, and convert this stream into a text in the format needed by the AIs.

It's as simple as doing buffer.toString('hex') and then parsing our the `\n` into an array.

This is why I asked you to collaborate with @sriram-vinnakota, the parsing of the document itself is excessively simple, what I wanted you to learn was to collaborate  to understand what is the output needed: 
does a single string delimited with commas and return line enough
```ts
const myDocument = "one cell, another cell, 0, 2\n1,1";
```
 or a list of strings 
```ts
const myDocument = [
  "one cell, another cell",
  "0, 2",
  "1,1"
];
```

Here is an example of a fully working flow:
https://github.com/A-star-logic/Ansearch/blob/main/packages/files-parsing/files-parsing.ts

and the code I was expecting (see, very simple) https://github.com/A-star-logic/Ansearch/blob/main/packages/files-parsing/files-parsing-text.ts

xichen-ansearch commented:
Hi,

I’ve made the following updates based on the previous review:

- Implemented CSV file parsing functionality.
- Added corresponding unit test for CSV file parsing.
- Fixed the incorrect file path in src/core/tests/e2e/core-extractor-e2e.test.ts.

Mael-Abgrall commented on src/parser/parser.ee.ts:
```diff
@@ -80,6 +86,15 @@ export async function parseStream({
 
+    case 'text/csv': {
+      const hexString = binaryStream.toString('hex');
+      const linesHex = hexString.split('0a');
```
can you explain this? this is not straightforward, and I did not have this in the ansearch source code

Mael-Abgrall replied:
aka make a comment in your code

xichen-ansearch replied:
I initially used 'utf8'. However I checked the review before and you mentioned "It's as simple as doing buffer.toString('hex') and then parsing out the \n into an array ". So I used 'hex' encoding. And in hexadecimal, the newline character \n is represented as 0a.

I was wondering if there's a specific requirement for using 'hex' over 'utf8'

Mael-Abgrall replied:
there is not, this is a brainfart from me sorry

use utf8

Mael-Abgrall commented on src/parser/tests/parser-csv.test.ts:
```diff
@@ -0,0 +1,45 @@
+    mimeType: undefined,
+  });
+
+  const expected = `text,list,region
```
good! I'm happy to see you understood the difference between the mimetype and without it

although, you can make your code a little neater, by using only one `expectedOutput` rather than copy-pasting it

Mael-Abgrall commented on src/parser/tests/parser-csv.test.ts:
```diff
@@ -0,0 +1,45 @@
+import { parseStream } from '../parser.ee.js';
+
+const csvPath = path.join(import.meta.dirname, 'sampleFiles', 'test.csv');
+// eslint-disable-next-line security/detect-non-literal-fs-filename
```
be careful with that line.

nothing wrong here, but I want you to be aware that disabling eslint should be an exception, and carefully considered, especially security rules


I will disable this linter rule on test files, as this security issue will never arise on test files

Mael-Abgrall reviewed (changes requested):
Good job, please address the comment and it should be ok to merge

xichen-ansearch commented:
Hi,

I've changed the code based on the review above.

Mael-Abgrall reviewed (approved):
Good work, merging now
